src/main/java/bio/terra/stairway/Flight.java

```diff
@@ -154,6 +155,10 @@ private FlightStatus fly() throws InterruptedException {
       // Dismal failure - undo failed!
       context().setResult(undoResult);
 
+    } catch (InterruptedException ex) {
+      // Interrupted exception - we assume this means that the thread pool is shutting down and
+      // forcibly stopping all threads. We propagate the exception.
+      throw ex;
```

Comment:
If I'm understanding correctly, by throwing the error, we're saying that this failed for extraneous reasons, so it was not a step failure.
Just curious - What will happen when this flight is picked up again? Will it just retry this step? If that's the case, why not set the step status to STEP_RESULT_FAILURE_RETRY?

Reply:
Not exactly. The InterruptedException is used in the specific case where the thread pool is terminating all of the flight threads during pod termination. The exception is caught two frames above and handled there. About line 95.

The result is that the flight is disowned - this stairway instance cannot own it since it is terminating - and will be picked up by another stairway instance. It will start at the step that was interrupted. That was the bug before: the fight with restarting at the next step.

By design, all steps have to be idempotent.